packages/cozy-harvest-lib/src/components/TriggerLauncher.jsx

```diff
@@ -0,0 +1,150 @@
+import { accountsMutations } from '../connections/accounts'
+import { triggersMutations } from '../connections/triggers'
+
+import KonnectorJob from '../models/KonnectorJob'
+
+// Statuses
+const ERRORED = 'ERRORED'
+const IDLE = 'IDLE'
+const LOGIN_SUCCESS = 'LOGIN_SUCCESS'
+const TWO_FA_REQUEST = 'TWO_FA_REQUEST'
+const TWO_FA_MISMATCH = 'TWO_FA_MISMATCH'
+const PENDING = 'PENDING'
+const SUCCESS = 'SUCCESS'
+const RUNNING_TWOFA = 'RUNNING_TWOFA'
+
+/**
+ * Trigger Launcher renders its children with following props:
+ * * launch: Callback to launch the trigger
+ * * running: Boolean which indicates if the trigger is running
+ *
+ * ### Example
+ * ```js
+ * <TriggerLauncher trigger={trigger}>
+ *   {(launch, running) => (
+ *     <Button onClick={launch} disabled={running}
+ *    )}
+ * </TriggerLauncher>
+ * ```
+ *
+ * TODO: inject other props to indicates status (for example: loginSucceed),
+ * and use callbacks like `onLoginSuccess` to make this component usable in
+ * TriggerManager for example
+ */
+export class TriggerLauncher extends Component {
+  state = {
+    status: IDLE
+  }
+
+  constructor(props, context) {
+    super(props, context)
+
+    this.launch = this.launch.bind(this)
+    this.handleTwoFAModalDismiss = this.handleTwoFAModalDismiss.bind(this)
+    this.handle2FACode = this.handle2FACode.bind(this)
+    this.handleError = this.handleError.bind(this)
+    this.handleSuccess = this.handleSuccess.bind(this)
+    this.handleLoginSuccess = this.handleLoginSuccess.bind(this)
+    this.handleTwoFARequest = this.handleTwoFARequest.bind(this)
+  }
+
+  handleTwoFAModalDismiss() {
+    // TODO: Make the modale not closable, or offer possibility to re-open it.
+    this.setState({ showTwoFAModal: false })
+  }
+
+  handleError() {
+    this.setState({ status: ERRORED, showTwoFAModal: false })
+  }
+
+  handleSuccess() {
+    this.setState({ status: SUCCESS, showTwoFAModal: false })
+  }
+
+  handleLoginSuccess() {
+    this.setState({ status: LOGIN_SUCCESS })
+  }
+
+  handleTwoFARequest() {
+    this.setState({ status: TWO_FA_REQUEST, showTwoFAModal: true })
+  }
+
+  handleTwoFAMismatch() {
+    this.setState({ status: TWO_FA_MISMATCH, showTwoFAModal: true })
+  }
+
+  handle2FACode(code) {
+    this.setState({ status: RUNNING_TWOFA })
+    this.konnectorJob.sendTwoFACode(code)
+  }
+
+  async launch() {
+    const { trigger } = this.props
+    const { client } = this.context
+
+    this.setState({ status: PENDING })
+
+    this.konnectorJob = new KonnectorJob(client, trigger)
+    this.konnectorJob
+      .on('error', this.handleError)
+      .on('loginSuccess', this.handleLoginSuccess)
+      .on('success', this.handleSuccess)
+      .on('twoFARequest', this.handleTwoFARequest)
+      .on('twoFAMismatch', this.handleTwoFAMismatch)
+    this.konnectorJob.launch()
+  }
+
+  render() {
+    const { showTwoFAModal, status } = this.state
+    const { account, children, konnector, submitting } = this.props
```

Comment:
I agree. As `submitting` is also used in TriggerManager, I suggest to keep it now for consistence and rename all props at the same time. I made an issue: https://github.com/cozy/cozy-libs/issues/431.

Reply:
@CPatchane `account` and `konnector` are data which can be retrieved with trigger infos. As konnectorJob is already fetching the related `account`, I suggest to also fetch the related konnector in it.

Then we pass the whole konnectorJob to TwoFAModal or we get account and konnector from it.

We should try to limit the number of props which can be deduced with other ones now, otherwise we will soon have a new Cozy-Home into Harvest.

Reply:
See https://github.com/cozy/cozy-libs/pull/415/commits/fee5d4745535bc71fb6badaf248b2b3fcfb847d3